config/auth_databricks_oidc.go

```diff
@@ -0,0 +1,91 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"net/url"
+
+	"github.com/databricks/databricks-sdk-go/config/experimental/auth"
+	"github.com/databricks/databricks-sdk-go/credentials/u2m"
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Creates a new Databricks OIDC TokenSource.
+func NewDatabricksOIDCTokenSource(cfg *DatabricksOIDCTokenSourceConfig) auth.TokenSource {
+	return &databricksOIDCTokenSource{
+		cfg: cfg,
+	}
+}
+
+// Config for Databricks OIDC TokenSource.
+type DatabricksOIDCTokenSourceConfig struct {
+	// ClientID is the client ID of the Databricks OIDC application. For
+	// Databricks Service Principal, this is the Application ID of the Service Principal.
+	ClientID string
+	// [Optional] AccountID is the account ID of the Databricks Account.
+	// This is only used for Account level tokens.
+	AccountID string
+	// Host is the host of the Databricks cluster.
+	Host string
+	// TokenEndpointProvider is a function that returns the token endpoint for the Databricks OIDC application.
```

Comment:
slightly more concise?
```suggestion
	// TokenEndpointProvider returns the token endpoint for the Databricks OIDC application.
```